src/args.rs

```diff
@@ -0,0 +1,154 @@
+    pub version6: bool,
+
+    /// limit the number of concurrent clients that can be processed by a server;
+    /// any over this count will be immediately disconnected
+    #[arg(long, value_name = "number", default_value = "0")]
+    pub client_limit: usize,
+
+    /// run in client mode; value is the server's address
+    #[arg(short, long, value_name = "host", conflicts_with = "server")]
+    pub client: Option<std::net::IpAddr>,
+
+    /// run in reverse-mode (server sends, client receives)
+    #[arg(short = 'R', long)]
+    pub reverse: bool,
+
+    /// the format in which to deplay information (json, megabit/sec, megabyte/sec)
+    #[arg(
+        short,
+        long,
+        value_enum,
+        value_name = "format",
+        default_value = "megabit"
+    )]
+    pub format: Format,
+
+    /// use UDP rather than TCP
+    #[arg(short, long)]
+    pub udp: bool,
+
+    /// target bandwidth in bytes/sec; this value is applied to each stream,
+    /// with a default target of 1 megabit/second for all protocols (note: megabit, not mebibit);
+    /// the suffixes kKmMgG can also be used for xbit and xbyte, respectively
+    #[arg(short, long, default_value = "125000", value_name = "bytes/sec")]
+    pub bandwidth: String,
+
+    /// the time in seconds for which to transmit
+    #[arg(short, long, default_value = "10.0", value_name = "seconds")]
+    pub time: f64,
+
+    /// the interval at which to send batches of data, in seconds, between [0.0 and 1.0);
+    /// this is used to evenly spread packets out over time
+    #[arg(long, default_value = "0.05", value_name = "seconds")]
+    pub send_interval: f64,
+
+    /// length of the buffer to exchange; for TCP, this defaults to 32 kibibytes; for UDP, it's 1024 bytes
+    #[arg(
+        short,
+        long,
+        default_value = "32768",
+        default_value_if("udp", "true", Some("1024")),
+        value_name = "bytes"
+    )]
+    pub length: usize,
+
+    /// send buffer, in bytes (only supported on some platforms;
+    /// if set too small, a 'resource unavailable' error may occur;
+    /// affects TCP window-size)
+    #[arg(long, default_value = "0", value_name = "bytes")]
+    pub send_buffer: usize,
+
+    /// receive buffer, in bytes (only supported on some platforms;
+    /// if set too small, a 'resource unavailable' error may occur; affects TCP window-size)
+    #[arg(long, default_value = "0", value_name = "bytes")]
+    pub receive_buffer: usize,
+
+    /// the number of parallel data-streams to use
+    #[arg(short = 'P', long, value_name = "number", default_value = "1")]
+    pub parallel: usize,
+
+    /// omit a number of seconds from the start of calculations,
+    /// primarily to avoid including TCP ramp-up in averages;
+    /// using this option may result in disagreement between bytes sent and received,
+    /// since data can be in-flight across time-boundaries
+    #[arg(short, long, default_value = "0", value_name = "seconds")]
```

Comment:
`short = 'O'` is required here. 

The unit-tests, and many real-world use-cases are broken without it, so this seems like a (minor) refactoring regression.